### PR TITLE
full implementation of the renewables implementation (Sullivan et al.)

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [ ] Description in RELEASE_NOTES.md Added

# Adding to RELEASE_NOTES.md

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

### CODE REVIEW

File: tests/test_tutorials.py [hunk 1]
```diff
@@ -53,12 +53,12 @@ def _notebook_run(path, kernel=None, capsys=None):
 
 @pytest.mark.skipif(not jupyter_installed, reason='requires Jupyter Notebook to be installed')
 def test_westeros(capsys):
-    fname = os.path.join(westeros_path, 'westeros.ipynb')
+    fname = os.path.join(westeros_path, 'westeros_part1.ipynb')
```
Comment (gidden) [inline tests/test_tutorials.py]: should this be baseline?

File: message_ix/model/MESSAGE/model_core.gms [hunk 2]
```diff
@@ -892,9 +894,12 @@ ADDON_ACTIVITY_LO(node,type_addon,year,mode,time)..
 * This constraint is only active if :math:`peak\_load\_factor_{n,c,l,y,h}` is defined.
 * The auxiliary variable :math:`COMMODITY\_USE_{n,c,l,y}` is only required
 * for the equations :math:`FIRM\_CAPACITY\_CONSTRAINT` and :math:`FIRM\_CAPACITY\_SHARE`.
+* WENN ENTWEDER PEAK LOAD ODER FLEXIBILITY DEMAND DA IST MUSS DIE COMM_USE BERECHNET WERDEN
```
Comment (gidden) [inline message_ix/model/MESSAGE/model_core.gms]: auf Englisch bitte

File: message_ix/model/MESSAGE/model_core.gms [hunk 3]
```diff
@@ -903,6 +908,55 @@ COMMODITY_USE_LEVEL(node,commodity,level,year,time)$( peak_load_factor(node,comm
                 * ACT(location,tec,vintage,year,mode,time2) )
 ;
 
+Positive Variable
+    ACT_SHARE(node,tec,vintage,year_all,commodity,level,time,rating)
+;
+
+* WENN ENTWEDER PEAK LOAD ODER FLEXIBILITY DEMAND DA IST UND RATING MUSS ?BER ACT_SHARE DIE BIN SIZE BERECHNET WERDEN
```
Comment (gidden) [inline message_ix/model/MESSAGE/model_core.gms]: auf Englisch bitte

File: message_ix/model/MESSAGE/model_core.gms [hunk 4]
```diff
@@ -903,6 +908,55 @@ COMMODITY_USE_LEVEL(node,commodity,level,year,time)$( peak_load_factor(node,comm
...
+ACTIVITY_SHARE_BIN(node,tec,year,commodity,level,time,rating)$(
+         rating_bin(node,tec,year,commodity,level,time,rating) )..
+   sum(vintage$(sum(mode,map_tec_act(node,tec,year,mode,time)) AND map_tec_lifetime(node,tec,vintage,year)),
+            ACT_SHARE(node,tec,vintage,year,commodity,level,time,rating) )
+    =L= rating_bin(node,tec,year,commodity,level,time,rating) * COMMODITY_USE(node,commodity,level,year)
+;
+
+ACTIVITY_SHARE_TOTAL(node,tec,vintage,year,commodity,level,time)$(
```
Comment (gidden) [inline message_ix/model/MESSAGE/model_core.gms]: there's a number of new constraints here. do we want docs for them or no?

**Comment by gidden** [comment]
also, we should add the new tutorials to `test_tutorials.py`

**Comment by gidden** [comment]
changed base branch to `hackathon-prs`

**Comment by danielhuppmann** [comment]
added new tutorial notebooks to unit tests

**Comment by gidden** [comment]
çlosing in favor of #92... had too hard of a time trying to get the merge issue to work

**Pull request closed without merging**